File: server/models/payment.js
const mongoose = require("mongoose");

const PaymentSchema = new mongoose.Schema({
  paidAmount: {
    type: Number,
    required: true,
  },
  leftAmount: {
    type: Number,
    required: true,
  },
  paidTo: {
    type: String, // phone,
  },
  paidBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Student",
  },
  lastUpdate: {
    type: Date,
    default: Date.now(),
  },
  dueDate: {
    type: Date,
  },
  originalDueDate: {
    type: Date,
  },
  statusChange: {
    type: Date,
    default: Date.now(),
  },
  paymentStatus: {
    type: String,
    enum: ["unpaid", "paid", "late", "postponed"],
    default: "unpaid",
  },
  installment: {
    //bu kaçıncı taksit
    type: Number,
    required: true,
  },
  totalInstallment: {
    //sevise kaç ay yazıldı
    type: Number,
    required: true,
  },
  additional: {
    type: Boolean,
    default: false
  },
  originalAmount: {
    type: Number,
    required: true,
  }
});

module.exports = Payment = mongoose.model("Payment", PaymentSchema);

File: socoolbus/lib/models/payment.dart
import 'package:my_app/my_enum.dart';

class Payment {
  final String paymentId;
  final int totalAmount;
  final String paidBy;
  final PaymentStatus status;
  final String paymentNumber;
  final int paidAmount;
  final int leftAmount;

  const Payment(
      {required this.paymentId,
      required this.totalAmount,
      required this.paidBy,
      required this.status,
      required this.paymentNumber,
      required this.paidAmount,
      required this.leftAmount});

  factory Payment.fromJson(Map<dynamic, dynamic> json) {
    PaymentStatus temp;
    if(json['paymentStatus'] == 'paid'){
        temp = PaymentStatus.paid;
    }
    else if(json['paymentStatus'] == 'unpaid'){
        temp = PaymentStatus.unpaid;
    }else if(json['paymentStatus'] == 'postponed'){
        temp = PaymentStatus.postponed;
    } else if(json['paymentStatus'] == 'late'){
        temp = PaymentStatus.late;
    }else {
        temp = PaymentStatus.requested;
    }
    return Payment(
      paymentId: json['_id'],
      totalAmount: (json['leftAmount'] + json['paidAmount']).toInt(),
      leftAmount: json['leftAmount'].toInt(),
      paidBy: json['paidBy'],
      status: temp,
      paymentNumber: json['paymentNumber'],
      paidAmount: json['paidAmount'].toInt(),
    );
  }
}
